### PR TITLE
Add minimal support for Dash's special transactions

### PR DESCRIPTION
**Pull request by UdjinM6**
For more info about special transactions pls see https://github.com/dashpay/dips/blob/master/dip-0002.md

### CODE REVIEW

**Comment by argjv** [comment]
Thanks for the PR @UdjinM6 it looks promising. I'll be doing some testing this week and add my review.
Thanks!

**Comment by paddyquinn** [comment]
Hi @UdjinM6. Thank you for the PR. We require all of our commits to the master repo to be gpg signed so we could not merge your PR to master. I have retargeted and merged it to a development branch. From there I have rebased against master and amended your commit with my signature (keeping you as the author). See #39 for the PR to master. Thanks again for your contribution!

File: src/transaction.js [hunk 1]
```diff
@@ -986,6 +1022,9 @@ Transaction.prototype.__toBuffer = function (buffer, initialOffset, __allowWitne
     var mask = (this.overwintered ? 1 : 0)
     writeInt32(this.version | (mask << 31))  // Set overwinter bit
     writeUInt32(this.versionGroupId)
+  } else if (this.isDashSpecialTransaction()) {
+    writeInt16(this.version)
+    writeInt16(this.type)
```
Comment (paddyquinn) [inline src/transaction.js]: It seems that these two writes should be `writeUInt16` per [DIP002](https://github.com/dashpay/dips/blob/master/dip-0002.md#special-transactions). This is already merged but I have another PR coming shortly to add special transaction signing to this library. I will include this fix there.

File: src/transaction.js [hunk 2]
```diff
@@ -47,6 +47,11 @@ function Transaction (network = networks.bitcoin) {
     this.vShieldedOutput = []
     this.bindingSig = 0
   }
+  if (coins.isDash(network)) {
+    // Dash version = 3
+    this.type = 0
+    this.extra_payload = 0
```
Comment (paddyquinn) [inline src/transaction.js]: We use camelCase in this lib so this new field should follow that convention. Will also fix in the future PR mentioned below. Should have caught this in review.

Comment (paddyquinn) [reply src/transaction.js]: The extra payload should be set to `Buffer(0)` for type consistency. At this point I am just taking notes for my next PR :) Review will be more thorough next time. Apologies.

File: src/transaction.js (hunk 1 above)
Comment (paddyquinn) [reply src/transaction.js]: See #40

File: src/transaction.js (hunk 2 above)
Comment (paddyquinn) [reply src/transaction.js]: See #40

**Pull request merged**